### PR TITLE
Adding https to the api

### PR DESCRIPTION
**Pull request by crazy7bird**
Hello, i need https for my api, so i code it, this is the result.
If you want to use https you have to start node init.js wi th sudo.
For get certificate https, i follow this tuto : https://www.sitepoint.com/how-to-use-ssltls-with-node-js/
By default api is on port 8117 and https api is on port 8118. 

If you need help for explanations or setting https feel free to ask !

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -1,8 +1,21 @@
 var fs = require('fs')
 var http = require('http')
+var https = require('https')
 var url = require('url')
 var zlib = require('zlib')
 
+
+//TODO find a way for no sudo..
```
Comment (brandonlehmann) [inline lib/api.js]: @crazy7bird What about this requires sudo? Are you referring to keeping the keys with everything else?

Have you taken a look at letsencrypt ?

**Comment by brandonlehmann** [comment]
I've re-targeted this PR for the development branch.

File: lib/api.js (hunk 1 above)
Comment (crazy7bird) [reply lib/api.js]: i use letsencrypt, but i just can't access to keys if i dont run node with sudo.
Maybe i do somthing wrong, but for the moment this solution work with me.
when i got time i'll try to do something without sudo.

**Comment by crazy7bird** [comment]
@DiscoTim, cause i don't know theses solutions ^^.

**Comment by brandonlehmann** [comment]
No movement from author in two weeks. Closing PR for now.

**Pull request closed without merging**